src/main/java/org/folio/circulation/domain/policy/lostitem/LostItemPolicy.java

```diff
@@ -1,51 +1,73 @@
 package org.folio.circulation.domain.policy.lostitem;
 
+import static java.util.Optional.ofNullable;
+import static org.folio.circulation.domain.policy.lostitem.ChargeAmountType.SET_COST;
+import static org.folio.circulation.domain.policy.lostitem.ChargeAmountType.forValue;
 import static org.folio.circulation.support.JsonPropertyFetcher.getBigDecimalProperty;
 import static org.folio.circulation.support.JsonPropertyFetcher.getBooleanProperty;
 import static org.folio.circulation.support.JsonPropertyFetcher.getObjectProperty;
 import static org.folio.circulation.support.JsonPropertyFetcher.getProperty;
 
 import java.math.BigDecimal;
+import java.util.Optional;
 
 import org.folio.circulation.domain.policy.Policy;
 
 import io.vertx.core.json.JsonObject;
 
 public class LostItemPolicy extends Policy {
-  private final ChargeAmount chargeAmountItem;
+  private final ItemFee setCostChargeFee;
   private final boolean chargeAmountItemPatron;
-  private final BigDecimal lostItemProcessingFee;
+  private final ItemFee processingChargeFee;
 
-  private LostItemPolicy(String id, String name, ChargeAmount chargeAmountItem,
-    boolean chargeAmountItemPatron, BigDecimal lostItemProcessingFee) {
+  private LostItemPolicy(String id, String name, ItemFee setCostChargeFee,
+    boolean chargeAmountItemPatron, ItemFee processingChargeFee) {
 
     super(id, name);
-    this.chargeAmountItem = chargeAmountItem;
+    this.setCostChargeFee = setCostChargeFee;
     this.chargeAmountItemPatron = chargeAmountItemPatron;
-    this.lostItemProcessingFee = lostItemProcessingFee;
+    this.processingChargeFee = processingChargeFee;
   }
 
   public static LostItemPolicy from(JsonObject lostItemPolicy) {
     return new LostItemPolicy(
       getProperty(lostItemPolicy, "id"),
       getProperty(lostItemPolicy, "name"),
-      ChargeAmount.from(getObjectProperty(lostItemPolicy, "chargeAmountItem")),
+      getSetCostFee(lostItemPolicy),
       getBooleanProperty(lostItemPolicy, "chargeAmountItemPatron"),
```

Comment:
What does this boolean represent?

Reply:
Whether to charge a processing fee or not.

Reply:
Does this need to be a field in the policy. Could it instead be used to determine if there is an item processing fee?

Am I understanding it correctly, that even if an item processing fee is defined, if this flag is false, then no fee is ever issued?

Reply:
Yes, correct. Sure it can.